lib/streamlit/widgets.py

```diff
@@ -55,13 +55,15 @@ def coalesce_widget_states(
     return coalesced
 
 
-class Widgets(object):
+class WidgetStateManager(object):
     def __init__(self):
-        self._state = {}  # type: Dict[str, WidgetState]
+        self._widget_states = {}  # type: Dict[str, WidgetState]
+        self._widget_callbacks = {}  # type: Dict[str, Callable[..., None]]
+        self._widget_deserializers = {}  # type: Dict[str, Callable[..., Any]]
```

Comment:
Will update the type comment. Hmmm...I'll keep this in mind. Dataclasses look like they are introduced in 3.7, and we _technically_ support 3.6....probably not for long. I think the widget state management could probably be better organized in some ways since they are kinda added independently (setting the state, the callback/deserializer are done in separate places.)